core/base/src/TString.cxx

```diff
@@ -1452,6 +1452,46 @@ TString operator+(const TString &s, char c)
 ////////////////////////////////////////////////////////////////////////////////
 /// Add integer to string.
 
+TString operator+(const TString &s, Short_t i)
+{
+    char si[32];
+    snprintf(si, sizeof(si), "%ld", i);
```

Comment:
`i` is not a `long`, thus the format spec is wrong. See other occurrences of the same issue.